dashboard/app/controllers/sections_controller.rb

```diff
@@ -46,6 +46,15 @@ def log_in
     end
   end
 
+  # Get the set of sections that the current user is enrolled in.
+  def membership
+    unless current_user
+      return head :forbidden
+    end
```

Comment:
This should be handled by `CanCan`. You shouldn't need to explicitly do any authorization in the controller action in most cases.

Reply:
For instance, you can add `:membership` to the authorized actions for users with sections here: https://github.com/code-dot-org/code-dot-org/blob/5c343be37cb30023cded2116b10ceed22e655e11/dashboard/app/models/ability.rb#L72

```ruby
can [:index, :membership], Section, user_id: user.id
```

Then this existing [load_and_authorize_resource](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/api/v1/sections_controller.rb#L4) call will handle auth and loading. The only difference is it will redirect to sign in rather than returning a forbidden when not logged in. I presume that wasn't a specific requirement.

Then in your action, `@sections` will be pre-populated with all that user's sections.

Reply:
so the resulting function would just be:

```
def membership
  render json: {sections: current_user.sections_as_student}
end
```

or should i still do a `.nil?` (or equivalent) check on `current_user`?

Reply:
Though I just noticed you're looking for `sections_as_student`. The above will give sections the user owns. For student sections, you'll want a new ability.

Reply:
Close, it will need to be `@sections` (member set by CanCan in a before_action callback behind the scenes, not local `sections` which doesn't exist). Also you'll need to add the correct ability to scope sections_as_student.

Reply:
ahh, i see. i'll make that update. thanks for your help!